Automattic/wp-calypso
# Checkout: Change cached contact details to use useQuery instead of Redux

## Proposed Changes

This PR refactors the `useCachedContactDetails()` to use `useQuery` instead of the Redux state for cached contact details. This will help decouple checkout from Redux so it can more easily be used outside of calypso.

As part of this process, this PR also removes contact details from Redux state. This was relatively easy because there are only two places contact data is used: the domain management form, which uses a separate Redux Whois data store, and checkout, which uses a "managed" contact form where the data is kept in a separate `@wordpress/data` store. The only thing actually using the removed data was the hook which prefills the contact fields in checkout.

To clarify some of this removal, this PR refactors the ccTLD contact forms (.ca, .fr, and .uk) into TypeScript and removes all the parts that were unused.

## Testing Instructions

- Use an account which has cached contact details. Verify that, without this PR, when you visit checkout the billing info step gets auto-completed.
- On this branch, verify that when you visit checkout the billing info step gets auto-completed.

Because of the ccTLD refactors, it's also important to test that they still work.

- Add a .ca, .fr, or .uk domain to your shopping cart.
- Visit checkout and fill out the contact details step, including the additional ccTLD fields.
- Click to complete the contact details step.
- Verify there are no errors. **NOTE:** ccTLD form fields are not pre-populated in checkout before or after this diff.
